### ISSUE TITLE
HTML code field just throws everything in the trash when sending email manually to a contact

### ISSUE THREAD

**Issue by andersonjeccel** [mautic/mautic#13987]
### Mautic Version

5.0.x series

### Way of installing

I cloned the Mautic repo (not recommended or supported)

### PHP version

8.3.9

### What browsers are you seeing the problem on?

Chrome

### What happened?

Every time I try to manually send an email to a contact, the system simply deletes all the content that was entered in the HTML code field.
This behavior happens specifically when clicking the send button where an error message is returned on the screen warning of the need to fill in the field's contents, as if the system was not saving what I wrote.

https://github.com/user-attachments/assets/3ad762b6-5571-447b-99a5-4f0d57d2271c

### How can we reproduce this issue?

1. Open Contacts
2. Create new, filling necessary data and email
3. Save & Close
4. Click on Send email
5. Fill the HTML section, leaving subject blank
6. Click to Send
7. Everything you wrote on the HTML field will go away

### Relevant log output

```shell
No logs
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by andersonjeccel** [comment]
@abhisekmazumdar Could you check this one?

**Comment by BlockchainViper** [comment]
Hi, I would like to work on this.

**Comment by andersonjeccel** [comment]
@Emmanueltech Feel free to start!

**Comment by abhisekmazumdar** [comment]
> @abhisekmazumdar Could you check this one?

Sorry for the late reply. Last week I was away on holidays.

I see the issue here. CKEditor takes some time to convert the given text to HTML. You can observe this when we type, and the source button gets disabled. It becomes enabled when it replaces the text with HTML.

So if you wait for a second after typing and then try to send the mail, it should work.
I will try to investigate and provide some more details for @Emmanueltech.

**Comment by andersonjeccel** [comment]
Perhaps we could have a message appearing below the field, like "Saving..." while the Send button is disabled until it's ready

**Comment by abhisekmazumdar** [comment]
Yes, that sounds like the right thing to do.

**Comment by abhisekmazumdar** [comment]
@Emmanueltech Are you still working on this? If not, please un-assign. I can pick this up

**Comment by BlockchainViper** [comment]
@abhisekmazumdar. Done. I am not feeling alright. Feel free to work on it.

**Comment by andersonjeccel** [comment]
@abhisekmazumdar Probably we already have the necessary JS code to make the disabled state for buttons based on Bootstrap v3.4 (using mQuery to add a .disabled class)

It might be worth it to [check their docs](https://getbootstrap.com/docs/3.4/) to know how to use 

Regarding styles for the Saving message:
- Use the `.help-block` class to achieve a font-size of 12px
- And `.text-muted` to make it a secondary text (gray instead of black)

### Bug fixes are nice to go against the 5.1 branch

**Comment by abhisekmazumdar** [comment]
Thank you @andersonjeccel for suggestion and also guiding me that I should be looking to write this script over:

- app/bundles/EmailBundle/Assets/js/email.js
- app/bundles/EmailBundle/Assets/js/config.js

But I issue I'm facing is no on change or any other event is working for me to hook into the ck editor content changes. I'm sure I'm looking at it wrong way. So here what I have did so far:

<img width="1266" alt="image" src="https://github.com/user-attachments/assets/92f2fef5-a7a8-49c0-9c57-6e77ebd74a93">

I understand that the text area is coming from: https://github.com/mautic/mautic/blob/5.x/app/bundles/LeadBundle/Form/Type/EmailType.php#L102-L114

And somehow CKEditor is taking the inputs and adding the HTML format content into the textarea. So I thought to add an onchange callback and make the send button disabled and help text, but it didn't work out.

Not sure which event or which places I should consider looking more into.

**Comment by andersonjeccel** [comment]
@abhisekmazumdar I believe it's time to call Rembrand

**Comment by LordRembo** [comment]
@abhisekmazumdar You sure the problem is there is a delay in processing the CKE field before submit? I can't get the form to send source code content, no matter how long I wait. Only text gets submitted.

**Comment by abhisekmazumdar** [comment]
This is what I see when I  try it using gitpop and ddev locally
![Recording 2024-08-02 at 17 02 53](https://github.com/user-attachments/assets/0595aadd-7178-40af-ab73-96bf2e0b63c4)

**Comment by LordRembo** [comment]
@abhisekmazumdar Right, yeah, I get it. I'm trying to get the issues straight in my head and if I'm understanding everything correctly. Seems there's 2 issues then:

- When adding content in CKE (rich text), there is a delay before the content is parsed and made available for the form. The proposal was to look for a way to disable the submit until that is done. So if you're fast to submit, you either can't submit (error says the message is empty because the content was not yet registered) or only part of your text is submitted (your last typing before submit was not yet parsed). 
- The other problem, is that if you click the 'Source code' button in the CKE, nothing you type in there is registered at all. You can only work around that by clicking the 'Source code' button again, forcing the content to be read as 'normal' CKE content. But that's not an actual fix.

I'm having a look at both problems but similarly to you, having a tough time tracing what is happening where. Will put in a bit more time today.

**Comment by abhisekmazumdar** [comment]
Okay, I understand. The way I see the "Source Code" button is that it's not something that's usually needed to enabled. I see it as a source editing mode that helps make any changes that the CKEditor made HTML. So it's always necessary to remove the source editing mode by clicking on the same button so that the CKEditor can convert it to HTML.

I would suggest that when the source editing mode is on, the send button can be disabled as well.

**Comment by LordRembo** [comment]
I put in a fair bit of time but I also don't see how we can toggle the `disabled` state on the submit button, depending on the updated content from CKE.

What I'm looking for is:

- disable the 'Send' button when textarea with ~~ID `lead_quickemail_body`~~ with CKE exists and is empty
- disable the 'Send' button when the CKE field has `focus`
- enable the 'Send' button when the CKE field goes out of focus (`onBlur`) and textarea value changes
    - so listen to / monitor the textarea value (the content gets updated whenever you make changes in CKE)
    - OR maybe [CKE's event handlers ](https://ckeditor.com/docs/ckeditor5/latest/framework/deep-dive/event-system.html)can let you know when the finish updating the textarea

**Comment by LordRembo** [comment]
I think most (or even all) of the changes need to happen in `app/bundles/CoreBundle/Assets/js/11.editor.js`, in the `Mautic.InitCkEditor` function. This is where you can do stuff based on CKE fields and events.

**Comment by abhisekmazumdar** [comment]
Unassigning this for me for now. I'm not getting time to dive into this
